FilterViewController.m

```diff
@@ -26,6 +26,13 @@ - (void)viewDidLoad {
     self.currentSubViewControllerIndex = 0;
 }
 
+- (void)viewDidDisappear:(BOOL)animated{
+    
+    [super viewDidDisappear:animated];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self name:UIDeviceOrientationDidChangeNotification object:nil];
```

Comment:
for every remove statement you need an add statement. viewDidDisappear is called more often than viewDidLoad so you will likely end up having periods of time that this observer isn't active when it should be.

If you only need an observer to be active while the view is visible I would add it in viewWillAppear and remove it in viewDidDisappear.

If you need an observer to be active while the view is not visible I would add the observer in init or viewDidLoad (depending on whether it's callback needs to manipulate the view, if it does you need the view to be loaded before activating the observer) and I would remove it in the dealloc method (that method is called just before a class is destroyed.